packages/libs/react-ui/src/components/Button/Button.tsx

```diff
@@ -1,43 +1,92 @@
-import { colorVariants } from './Button.css';
+import {
+  buttonLoadingClass,
+  colorVariants,
+  iconLoadingClass,
+} from './Button.css';
+import { ButtonIcon } from './ButtonIcon';
 
+import { SystemIcon } from '@components/Icon';
+import cx from 'classnames';
 import React, { ButtonHTMLAttributes, FC } from 'react';
 
 export interface IButtonProps
   extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'as' | 'disabled'> {
   as?: 'button' | 'a';
-  onClick?: React.MouseEventHandler<HTMLButtonElement>;
-  href?: string;
+  variant?: keyof typeof colorVariants;
   children: React.ReactNode;
-  title?: string;
   disabled?: boolean;
-  color?: keyof typeof colorVariants;
+  href?: string;
+  icon?: keyof typeof SystemIcon;
+  iconAlign?: 'left' | 'right';
+  loading?: boolean;
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  target?: '_blank' | '_self';
+  title?: string;
 }
 
 export const Button: FC<IButtonProps> = ({
   as = 'button',
-  color = 'primary',
-  onClick,
-  href,
   children,
+  variant = 'primary',
+  href,
+  icon,
+  iconAlign,
```

Comment:
generally I think the icon will be aligned to the right. I think it would be beneficial to give this prop a default value so that most often this prop won't need to be set. What do you think?

Reply:
Yeah I noticed the same, good idea! Will change it.